### ISSUE TITLE
[Enhancement] Expose Resolve(Type) and Get(Type) methods on DependencyService to be used in a non generic way

### ISSUE THREAD

**Issue by danielmeza** [xamarin/Xamarin.Forms#9424]
## Summary
Expose a Resolve(Type) and Get(Type) methods on [DependencyService ](https://github.com/xamarin/Xamarin.Forms/blob/f0087cbf5af2b11cca6b5bc6977c1ea75da0dd40/Xamarin.Forms.Core/DependencyService.cs#L19)for used in a non generic way.

## API Changes

A consumer can now use the api like,

```csharp
Type someType ;//some type from a external source, like a MarkupExtension 
var instance = DependencyService.Resolve(someType); //new API method
```
same for Get<T> method

The methods will look like:
```csharp
public static object Resolve(Type serviceType, DependencyFetchTarget fallbackFetchTarget = DependencyFetchTarget.GlobalInstance);

public static object Get(Type serviceType, DependencyFetchTarget fallbackFetchTarget = DependencyFetchTarget.GlobalInstance);
```
We can convert the main methods in this and expose the generic version as extensions methods or included in the same class. 

This change will not affect any existing app.
## Intended Use Case
This is useful in cases where the generic parameter is not possible, like a MarkupExtension  it will allow to external libraries easy relay on the default service resolution mechanism provided by the framework, eg. 

An MarkupExtensions that provide a service to a given type can pass only the Type of the service and non the generic parameter, since the DependencyService.Resolve use the DependencyResolver and a user can change the default resolution strategy to use its how resolution strategy.

I know that this can be archive using reflection but, since this change non break any code can be implemented with safety, and avoid the payload of use reflection. 

## Related issues
Support for DependencyTemplateExtension https://github.com/xamarin/Xamarin.Forms/issues/9425

**Comment by Insire** [comment]
I'm looking forward to this, so i can use ``DependencyService`` as fallback for ``DryIoc``.

TLDR: I want to replace ``DependencyService`` in my .netstandard project and use ``DryIoc`` as primary IoC container, while still being able to use the platform specific registrations in the android and ios projects.

**Comment by jfversluis** [comment]
This will not happen anymore for Xamarin.Forms. If this is still important to you, make sure to check the [.NET MAUI repo](https://github.com/dotnet/maui) and see if it's already on the roadmap or open an issue with a detailed feature request. Thanks!
